src/main/java/org/sunbird/assessment/service/AssessmentServiceV2Impl.java

```diff
@@ -35,195 +40,372 @@ public class AssessmentServiceV2Impl implements AssessmentServiceV2 {
+                    if (!ObjectUtils.isEmpty(questionMapResponse) && Constants.OK.equalsIgnoreCase((String) questionMapResponse.get(Constants.RESPONSE_CODE))) {
+                        List<Map<String, Object>> questionMap = ((List<Map<String, Object>>) ((Map<String, Object>) questionMapResponse
+                                .get(Constants.RESULT)).get(Constants.QUESTIONS));
+                        for (Map<String, Object> question : questionMap) {
+                            if (!ObjectUtils.isEmpty(questionMap)) {
+                                Boolean isInsertedToRedis = redisCacheMgr.putCache(Constants.QUESTION_ID + question.get(Constants.IDENTIFIER), question);
+                                questionList.add(filterQuestionMapDetail(question));
+                            } else {
+                                errMsg = "Failed to get Question Details for Id: %s";
+                                logger.error(String.format("Failed to get Question Details for Id: %s", question.get(Constants.IDENTIFIER).toString()));
+                                break;
+                            }
                         }
+                    } else {
+                        logger.error(String.format("Failed to get Question Details from the Question List API for the IDs: %s", newIdentifierList.toString()));
                     }
                 }
+                if (!questionList.isEmpty() && errMsg.isEmpty()) {
+                    response.getResult().put(Constants.QUESTIONS, questionList);
+                } else {
+                    errMsg = "The list of questions are Empty/Invalid";
+                }
             }
-            return prepareQuestionResponse(questionList, questionList.size() > 0);
         } catch (Exception e) {
-            logger.error(e);
-            throw new ApplicationLogicError("REQUEST_COULD_NOT_BE_PROCESSED", e);
+            logger.error(String.format("Exception in %s : %s", "get Question List", e.getMessage()), e);
+            errMsg = "Failed to fetch the question list. Exception: " + e.getMessage();
         }
+        if (StringUtils.isNotBlank(errMsg)) {
+            response.getParams().setStatus(Constants.FAILED);
+            response.getParams().setErrmsg(errMsg);
+            response.setResponseCode(HttpStatus.BAD_REQUEST);
+        }
+        return response;
+
+    }
 
+    private String validateAuthTokenAndFetchUserId(String authUserToken) {
+        return RequestInterceptor.fetchUserIdFromAccessToken(authUserToken);
+    }
+
+    private String validateQuestionListAPI(Map<String, Object> requestBody, String authUserToken, String errMsg, List<String> identifierList) {
+        String userId = validateAuthTokenAndFetchUserId(authUserToken);
+        if (userId != null && requestBody.containsKey(Constants.ASSESSMENT_ID_KEY) && !StringUtils.isEmpty((String)requestBody.get(Constants.ASSESSMENT_ID_KEY))) {
```

Comment:
This is bad...
if userId is not null and requestBody doesn't contain AssessmentId - still the error message would be -- ""User Id doesn't exist! Please supply a valid auth token" -- why ??